tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,113 @@
+    await page.getByRole("button", { name: "Save Pet" }).click();
+
+    const tableStripedSelector = page.locator("table.table-striped").last().locator("tr td dd");
+    await expect(tableStripedSelector.first()).toHaveText("Tom");
+    await expect(tableStripedSelector.nth(1)).toHaveText("2014-05-02");
+    await expect(tableStripedSelector.last()).toHaveText("dog");
+
+    await page.getByRole("button", { name: "Delete Pet" }).last().click();
+
+    await expect(tableStripedSelector.first()).toBeEmpty();
+  });
+
+  test("Select the dates of visits and validate dates order", async ({page}) => {
+    await page.getByText("Jean Coleman").click();
+
+    const samanthaTableStripedSelector = page.locator("table.table-striped").last().locator("tr td dd");
+    const samanthaAddVisitButton = page.locator("table.table-striped").last().getByRole("button", { name: "Add Visit" });
+    await expect(samanthaTableStripedSelector.first()).toHaveText("Samantha");
+    await samanthaAddVisitButton.click();
+
+    await expect(page.locator("h2")).toHaveText("New Visit");
+
+    const newVisitTableStripedSelector = page.locator("table.table-striped").locator("tr td");
+    await expect(newVisitTableStripedSelector.first()).toHaveText("Samantha");
+    await expect(newVisitTableStripedSelector.last()).toHaveText("Jean Coleman");
+
+    const calendarIcon = page.getByLabel("Open calendar");
+    await calendarIcon.click();
+    let date = new Date();
+    const currentYear = date.getFullYear();
+    const currentMonth = date.getMonth() + 1;
+    const currentDay = date.getDate();
+    const currentDate = `${currentYear}/${currentMonth}/${currentDay}`;
+
+    const calendarTodayCell = page.locator('[class="mat-calendar-body-cell-content mat-focus-indicator mat-calendar-body-today"]');
+    await calendarTodayCell.click();
+
+    const dateInputField = await page.locator('input[name="date"]');
+    await expect(dateInputField).toHaveValue(currentDate);
+
+    const descriptionInputField = await page.locator("#description");
+    await descriptionInputField.fill("dermatologist visit");
+    const newVisitAddVisitButton = await page.getByRole("button", {name: "Add Visit"});
+    await newVisitAddVisitButton.click();
+
+    const dermatologistVisitDate = `${currentYear}-${currentMonth}-${currentDay}`;
+    const petVisitsRow = page.locator("table.table-condensed").last().locator("tr td");
+    await expect(petVisitsRow.first()).toHaveText(dermatologistVisitDate);
+
+    await samanthaAddVisitButton.click();
+
+    await calendarIcon.click();
+    let targetDate = new Date();
+    targetDate.setDate(targetDate.getDate() - 45);
+    const secondVisitYear = date.getFullYear();
+    const secondVisitMonth = date.getMonth() + 1;
+    const secondVisitDay = date.getDate();
+    const secondVisitDate = `${secondVisitYear}/${secondVisitMonth}/${secondVisitDay}`;
+
+    await calendarTodayCell.click();
+    await expect(dateInputField).toHaveValue(secondVisitDate);
```

Comment:
The logic here is not implemented correctly. Script selects the same date twice, instead of selecting a date 45 days in the past
<img width="792" alt="Petclinic Test 2024-12-21 10-02-38" src="https://github.com/user-attachments/assets/60474b2c-12e1-4803-8b38-2d8712ac3b1e" />